core/src/main/java/org/elasticsearch/index/query/WrapperQueryBuilder.java

```diff
@@ -134,4 +127,21 @@ protected int doHashCode() {
     protected boolean doEquals(WrapperQueryBuilder other) {
         return Arrays.equals(source, other.source);   // otherwise we compare pointers
     }
+
+    @Override
+    public QueryBuilder<?> rewrite(QueryRewriteContext context) throws IOException {
+        try (XContentParser qSourceParser = XContentFactory.xContent(source).createParser(source)) {
+            QueryParseContext parseContext = context.newParseContext();
+            parseContext.reset(qSourceParser);
+
+            final QueryBuilder<?> queryBuilder = parseContext.parseInnerQueryBuilder();
+            if (queryBuilder.boost() == DEFAULT_BOOST) {
+                queryBuilder.boost(boost());
+            }
+            if (queryName() != null) { // we inherit the name
+                queryBuilder.queryName(queryName());
+            }
```

Comment:
do we want to move the boost logic and query name to AbstractQueryBuilder somehow? I am always afraid of forgetting these tiny details.